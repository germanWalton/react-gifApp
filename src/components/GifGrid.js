import React from "react";
import useFetchGifs from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";
// import getGifs from "../helpers/getGifs";
import PropTypes from "prop-types";

const GifGrid = ({ category }) => {
  const { data:images, loading } = useFetchGifs(category);

  return (
    <>
      <hr className="mt-5"/>
      <h3 className='animate__animated animate__zoomIn mt-5 text-center p-3'>{category}</h3>

      {loading && <p className='animate__animated animate__flash'>loading...</p>}

      {
        <div className="card-grid ">
          {images.map(img => (
            <GifGridItem
                key={img.id}
                {...img}
            />
          ))}
        </div>
      }
    </>
  );
};

GifGrid.propTypes = {
  category:PropTypes.string.isRequired
}


export default GifGrid;
