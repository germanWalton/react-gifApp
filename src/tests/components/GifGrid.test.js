import React from "react";
import { shallow } from "enzyme";
import '@testing-library/jest-dom'
import GifGrid from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs")


describe('Pruebas en <GifGrid />', () => {

  const category = 'Dragon Ball';

  test('debe mostrar el componente correctamente', () => {
  
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })

  const wrapper = shallow( <GifGrid category={ category } />)
  expect(wrapper).toMatchSnapshot()
})


  test('debe mostrar items cuando se cargan las imagenes useFetchGifs', () => {

    const gifs = [{
      id: 'ABC',
      url: 'http://www.google.com',
      title:'Dragon Ball'
    }]

    useFetchGifs.mockReturnValue({
      data:gifs,
      loading:false
    })

    const wrapper = shallow(<GifGrid category={category} />)
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  
})  

})