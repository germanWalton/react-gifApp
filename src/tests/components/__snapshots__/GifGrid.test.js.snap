// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe mostrar el componente correctamente 1`] = `
<Fragment>
  <hr
    className="mt-5"
  />
  <h3
    className="animate__animated animate__zoomIn mt-5 text-center p-3"
  >
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading...
  </p>
  <div
    className="card-grid "
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrar items cuando se cargan las imagenes useFetchGifs 1`] = `
<Fragment>
  <hr
    className="mt-5"
  />
  <h3
    className="animate__animated animate__zoomIn mt-5 text-center p-3"
  >
    Dragon Ball
  </h3>
  <div
    className="card-grid "
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Dragon Ball"
      url="http://www.google.com"
    />
  </div>
</Fragment>
`;
